rpm-ostree: add page (#8277)
tldr-pages/tldr: pages/linux/rpm-ostree.md

# rpm-ostree

> A hybrid image/package system.
> Manage ostree deployments, package layers, filesystem overlays, and boot configuration.
> More information: <https://coreos.github.io/rpm-ostree/administrator-handbook/>.

- Show rpm-ostree deployments in the order they will appear in the bootloader:

`rpm-ostree status`

- Show packages which are outdated and can be updated:

`rpm-ostree upgrade --preview`

- Prepare a new ostree deployment with upgraded packages and reboot into the new deployment:

`rpm-ostree upgrade --reboot`

- Reboot into the previous ostree deployment:

`rpm-ostree rollback --reboot`

> - Install a package into a new ostree deployment and reboot into the new deployment:

Reviewer: Would this work?
```suggestion
- Install a package into a new ostree deployment and reboot into it:
```

Reviewer 2: Yeah, It sounds good.

Author: SGTM!

> `rpm-ostree install {{htop}} --reboot`

Reviewer 2: ```suggestion
`rpm-ostree install {{package}} --reboot`
```

Author: Whoops, shoulda caught this when I pulled this locally and dropped apply-live. Thanks!

# rpm-ostree

> A hybrid image/package system.
> Manage ostree deployments, package layers, filesystem overlays, and boot configuration.
> More information: <https://coreos.github.io/rpm-ostree/administrator-handbook/>.

> - Show rpm-ostree deployments in the order they'll appear in the bootloader:

Reviewer 3: ```suggestion
- Show rpm-ostree deployments in the order they will appear in the bootloader:
```

I feel they'll is a little too informal.

`rpm-ostree status`

> - Show package updates that will be performed on the next rpm-ostree upgrade:

Reviewer 3: ```suggestion
- Show packages which are outdated and can be updated:
```

Author: I like this a lot more, thanks!

`rpm-ostree upgrade --preview`

- Prepare a new ostree deployment with upgraded packages and reboot into the new deployment:

`rpm-ostree upgrade --reboot`

> - Roll back and reboot into the previous ostree deployment:

Reviewer 3: ```suggestion
- Reboot into the previous ostree deployment:
```

`rpm-ostree rollback --reboot`

> - Install a package into a new ostree deployment and apply that deployment to the live deployment:

Reviewer 3: ```suggestion
- Install a package into a new ostree deployment and apply that deployment to the current live deployment:
```

Reviewer: Not sure if it's still correct, but what about this
```suggestion
- Install a package into a new ostree deployment and apply it to the current live deployment:
```

Author: I'll stew on this for a bit, `rpm-ostree` is a pretty strange (and somewhat complex) beast and I'm having trouble coming up with language that's precise/accurate/approachable. This language is more clear but there's some complex behavior happening with filesystem overlays, and I _think_ there's some unique behavior with this on reboots that needs to be captured.

If I can't figure out a good way to explain this in the next couple of hours then I'll drop this example and open up a new PR for `rpm-ostree install --apply-live` so this PR doesn't get stuck on my indecision. 😁

Reviewer 2: > I'll stew on this for a bit, `rpm-ostree` is a pretty strange (and somewhat complex) beast and I'm having trouble coming up with language that's precise/accurate/approachable. This language is more clear but there's some complex behavior happening with filesystem overlays, and I _think_ there's some unique behavior with this on reboots that needs to be captured.
> 
> If I can't figure out a good way to explain this in the next couple of hours then I'll drop this example and open up a new PR for `rpm-ostree install --apply-live` so this PR doesn't get stuck on my indecision. 😁

Hey, no need to remove the example from this PR, I will try to come up with an example

Reviewer 2: So, We can remove this command from this page currently as mentioned in [comment](https://github.com/tldr-pages/tldr/pull/8277/files/223d1b30ef57429027481cc9c70283758971909c#r936174001).

Author: Yeah, let's avoid the complexity and drop it. `rpm-ostree install -A {{package}}` seems too risky to have as the first 8 commands we should be sharing with users.

> `rpm-ostree install <htop> --apply-live`

Reviewer 3: ```suggestion
`rpm-ostree install {{package}} --apply-live`
```

Author: Apparently I forgot tldr's syntax 🤦  thanks!

Author: Regarding examples, I saw this comment in the [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#naming) and might be misunderstanding it:

> Use an actual value rather than a generic placeholder where appropriate. For example, use `iostat {{2}}` rather than `iostat {{interval_in_secs}}`.

For this style recommendation, the `iostat` examples use `{{2}}` over `{{interval}}` because that's a reasonable default for all cases, whereas using `{{htop}}` in this example is a bit too specialized?

Reviewer: Many of those rules emerged from what feels right. `seconds` may not be the best example for "appropriate", because you can hardly do something wrong with it. But this rule is mostly for examples like `{{time}}`, `{{date}}`, `{{version}}` where you don't really know what it should look like. 

Also it's clear that `{{seconds}}` is a placeholder, while `{{2}}` is a concrete value. `htop` is quite popular, but it could still be misinterpreted as placeholder. `{{package}}` on the other side is clearly a placeholder and you can hardly do anything wrong with the syntax.

Author: @marchersimon got it, thanks for the clarification!

- Install a package into a new ostree deployment and reboot into the new deployment:

> `rpm-ostree install <htop> --reboot`

Reviewer 3: ```suggestion
`rpm-ostree install {{package}} --reboot`
```
